packages/@o3r/components/src/tools/placeholder/placeholder.component.ts

```diff
@@ -47,27 +48,39 @@ export class PlaceholderComponent implements OnInit, OnDestroy {
   constructor(private store: Store<PlaceholderTemplateStore>, private cd: ChangeDetectorRef) {
   }
 
+  private sendMessage(data: PlaceholderLoadingStatus) {
+    sendOtterMessage<PlaceholderLoadingStatusMessage>('placeholder-loading-status', data, false);
+  }
+
   /** @inheritdoc */
   public ngOnInit() {
     this.subscription.add(
       this.id$.pipe(
         distinctUntilChanged(),
-        switchMap((id) => this.store.select(selectPlaceholderRenderedTemplates(id)))
-      ).subscribe((templates) => {
-        if (templates) {
-          this.isPending = templates.isPending;
-          const orderedRenderedTemplates = templates.orderedRenderedTemplates;
-          if (!orderedRenderedTemplates || !orderedRenderedTemplates.length) {
-            this.template = undefined;
-          } else {
-            // Concatenates the list of templates
-            this.template = orderedRenderedTemplates.join('');
-          }
-        } else {
-          this.isPending = false;
+        switchMap((id) =>
+          this.store.select(selectSortedTemplates(id)).pipe(
+            map((placeholders) => ({
+              id,
+              orderedTemplates: placeholders?.orderedTemplates,
+              isPending: placeholders?.isPending
+            })),
+            distinctUntilChanged()
+          )
+        )
+      ).subscribe(({id, orderedTemplates, isPending}) => {
+        this.isPending = isPending;
+        if (!orderedTemplates?.length) {
           this.template = undefined;
+        } else {
+          // Concatenates the list of templates
+          this.template = orderedTemplates.map(placeholder => placeholder.renderedTemplate).join('');
         }
         this.cd.markForCheck();
+        this.sendMessage({
```

Comment:
are we sure that the template is present in the DOM when the message is sent ?
I feel like this could lead to some race condition errors

Reply:
You are absolutely right